crates/miden-objects/src/block/nullifier_tree.rs

```diff
@@ -257,7 +257,10 @@ where
     /// Returns an iterator over the nullifiers and their block numbers in the tree.
     pub fn entries(&self) -> impl Iterator<Item = (Nullifier, BlockNumber)> {
         self.smt.entries().map(|(nullifier, block_num)| {
-            (Nullifier::from(nullifier), nullifier_leaf_value_to_block_num(block_num))
+            (
+                Nullifier::new_unchecked(nullifier),
+                nullifier_leaf_value_to_block_num(block_num),
```

Comment:
FYI: Created an issue as a follow-up to this PR: https://github.com/0xMiden/miden-base/issues/2125.